### PR TITLE
Allow machine agents to connect to debuglog endpoint

### PR DESCRIPTION
**Pull request by babbageclunk**
Previously only users could request the debug log, but we want to use it as the source for the migration log transfer, so the machine agent needs to be able to request it as well. 

Added `httpContext.stateForRequestAuthenticatedTag`, which allows the calling code to say what kinds of entities should be allowed. Also generalised `apiserver.common.AuthEither` into `AuthAny`, which can combine any number of auth functions.

Moved the base debuglog tests into `debuglog_db_test.go` to make them easier to find - the file version of the debuglog is gone now.

QA steps:
Bootstrapped and checked that I could still use debug-log. It's hard to check that a machine agent can use it, but I'll be checking that in my logtrasfer testing.

### CODE REVIEW

**Comment by babbageclunk** [comment]
!!build!!

File: apiserver/debuglog_db_test.go [hunk 1]
```diff
@@ -3,16 +3,130 @@
...
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/testing/factory"
+)
 
-// debugLogDBSuite runs the common debuglog API tests when the db-log
-// feature flag is enabled. These tests are inherited from
-// debugLogBaseSuite.
 type debugLogDBSuite struct {
-	debugLogBaseSuite
+	authHTTPSuite
```
Comment (mjs) [inline apiserver/debuglog_db_test.go]: Thanks for cleaning this up. These reason for this is that /log used to be backed by a file on disk and the suites were split when logging to mongodb was added behind a feature flag. There's not need for the separation now.

File: apiserver/firewaller/firewaller.go [hunk 2]
```diff
@@ -56,8 +56,8 @@ func NewFirewallerAPI(
 	accessUnit := common.AuthFuncForTagKind(names.UnitTagKind)
 	accessService := common.AuthFuncForTagKind(names.ApplicationTagKind)
```
Comment (mjs) [inline apiserver/firewaller/firewaller.go]: Gah! This should be called accessApplication. Do you mind fixing?

File: apiserver/firewaller/firewaller.go [hunk 3]
```diff
@@ -56,8 +56,8 @@ func NewFirewallerAPI(
...
 	accessMachine := common.AuthFuncForTagKind(names.MachineTagKind)
-	accessUnitOrService := common.AuthEither(accessUnit, accessService)
-	accessUnitServiceOrMachine := common.AuthEither(accessUnitOrService, accessMachine)
+	accessUnitOrService := common.AuthAny(accessUnit, accessService)
+	accessUnitServiceOrMachine := common.AuthAny(accessUnitOrService, accessMachine)
```
Comment (mjs) [inline apiserver/firewaller/firewaller.go]: Same for these. s/Service/Application/

File: apiserver/debuglog_db_test.go (hunk 1 above)
Comment (babbageclunk) [inline apiserver/debuglog_db_test.go]: Yeah, from some of the comments and filenames I guessed that there had been a file version that had been ripped out.

File: apiserver/firewaller/firewaller.go (hunk 2 above)
Comment (babbageclunk) [inline apiserver/firewaller/firewaller.go]: Sure.

File: apiserver/firewaller/firewaller.go (hunk 3 above)
Comment (babbageclunk) [inline apiserver/firewaller/firewaller.go]: Done - it turned out that this was actually doing Either(Either(unit, application), machine), so I changed it to Any(unit, application, machine) instead.

**Comment by babbageclunk** [comment]
$$merge$$

**Pull request merged**